src/main/java/eu/europeana/api/recommend/util/RecommendServiceUtils.java

```diff
@@ -0,0 +1,170 @@
+        s.append("&rows=").append(recordIds.length)
+                .append("&profile=minimal")
+                .append("&wskey=").append(wskey);
+        return s.toString();
+    }
+
+    /**
+     * Generates Entity Api search query to fetch all skos:preflabel and skos:altlabel
+     * for the given entity uri.
+     *
+     * example : ?query=entity_uri:%22<entityUri>%22&fl=skos_prefLabel.*,skos_altLabel.*&wskey=
+     * @param entityUri
+     * @param wskey
+     * @return
+     */
+    public String entityApiSearchQuery(String entityUri, String wskey) {
+        StringBuilder query = new StringBuilder("?query=entity_uri:");
+        query.append("\"" + entityUri + "\"");
+        query.append("&fl=skos_prefLabel.*,skos_altLabel.*");
+        query.append("&").append("wskey=").append(wskey);
+        return query.toString();
+    }
+
+    /**
+     * Generates set Api search query to fetch entity set items
+     * for the given entity uri.
+     *
+     * example : ?query=type:EntityBestItemsSet&f&qf=subject:<entityUri>&pageSize=<pageSize>&profile=standard&wskey=
+     * @param entityUri
+     * @param pageSize
+     * @param wskey
+     * @return
+     */
+    public String setApiSearchQuery(String entityUri, String pageSize, String wskey) {
+        StringBuilder query = new StringBuilder("?query=type:EntityBestItemsSet");
+        query.append("&qf=subject:").append(entityUri);
+        query.append("&pageSize=").append(pageSize);
+        query.append("&profile=standard");
+        query.append("&").append("wskey=").append(wskey);
+        return query.toString();
+    }
+
+    // ---------------------- Temporary functionality below to mock entity recommendations
+    private String generateEntitySearchQuery(String entityFullId, int maxResults, String wskey) {
+        StringBuilder s = new StringBuilder(50)
+                .append("?query=text:\"").append(entityFullId).append("\"")
+                .append("&rows=").append(maxResults)
+                .append("&sort=random")
+                .append("&profile=minimal")
+                .append("&wskey=").append(wskey);
+        return s.toString();
+    }
+
+    /**
+     * extracts items from set api response
+     * @param jsonObject
+     * @return
+     */
+    public List<String> extractItems(JSONObject jsonObject) {
+        List<String> items = new ArrayList<>();
+        try {
+            JSONArray setsList = jsonObject.getJSONArray(ITEMS_FIELD);
+            for(int i =0; i < setsList.length(); i++) {
+                JSONObject set = (JSONObject) setsList.get(i);
+                JSONArray itemList =  set.getJSONArray(ITEMS_FIELD);
+                for(int j =0; j < itemList.length(); j++) {
+                 items.add(StringUtils.substringAfter((String) itemList.get(j), "http://data.europeana.eu/item"));
+                }
+            }
+        } catch (JSONException e) {
+            LOG.error("Error fetching the objects from set api response. {}", e.getMessage());
```

Comment:
SQ issue: "Log or rethrow this exception". Either we should include the entire error and not only the message, or we should add @SuppressWarnings, but add a good java comment motivation why we are doing so.

BTW there's a saying/good programming practice that says "Throw early, catch late", so we could also propagate the error and catch it in the service instead.

Reply:
will catch in service instead.